src/app/clusters/general_diagnostics_server/general_diagnostics_server.cpp

```diff
@@ -219,6 +251,37 @@ class GeneralDiagnosticsDelegate : public DeviceLayer::ConnectivityManagerDelega
         ChipLogProgress(Zcl, "GeneralDiagnosticsDelegate: OnHardwareFaultsDetected");
 
         ReportAttributeOnAllEndpoints(GeneralDiagnostics::Attributes::ActiveRadioFaults::Id);
+
+        for (uint16_t index = 0; index < emberAfEndpointCount(); index++)
+        {
+            if (emberAfEndpointIndexIsEnabled(index))
+            {
+                EndpointId endpointId = emberAfEndpointFromIndex(index);
+
+                if (emberAfContainsServer(endpointId, GeneralDiagnostics::Id))
+                {
+                    // If General Diagnostics cluster is implemented on this endpoint
+                    MatterReportingAttributeChangeCallback(endpointId, GeneralDiagnostics::Id,
+                                                           GeneralDiagnostics::Attributes::ActiveRadioFaults::Id);
+
+                    // Record RadioFault event
+                    EventNumber eventNumber;
+                    EventOptions eventOptions(System::SystemClock().GetMonotonicTimestamp(), EventOptions::Type::kUrgent);
```

Comment:
Actually you don't need to setup time here, in event management, it would automatically setup the right epoch or system based upon system configuration.

Reply:
https://github.com/project-chip/connectedhomeip/pull/12711
just post this PR, then you would not have the above confusion. sorry for the inconvenience.